src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/QuickSurfaceMesh.cpp

```diff
@@ -984,22 +984,27 @@ void QuickSurfaceMesh::createNodesAndTriangles(std::vector<MeshIndexType>& m_Nod
   triangleGeom->getFaceAttributeMatrix()->resizeTuples({triangleCount});
   triangleGeom->getVertexAttributeMatrix()->resizeTuples(tDims);
 
-  auto& faceLabels = m_DataStructure.getDataRefAs<Int32Array>(m_Inputs->pFaceLabelsDataPath);
-  linkedGeometryData.addFaceData(m_Inputs->pFaceLabelsDataPath);
-  linkedGeometryData.addVertexData(m_Inputs->pNodeTypesDataPath);
+  auto& faceLabels = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FaceLabelsDataPath);
+  linkedGeometryData.addFaceData(m_InputValues->FaceLabelsDataPath);
+
+  // Remove and then insert a properly sized int8 for the NodeTypes
+  m_DataStructure.removeData(m_InputValues->NodeTypesDataPath);
+  Result<> nodeTypeResult = complex::CreateArray<int8_t>(m_DataStructure, {nodeCount}, {1}, m_InputValues->NodeTypesDataPath, IDataAction::Mode::Execute);
+  Int32Array& nodeTypes = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FaceLabelsDataPath);
+  linkedGeometryData.addVertexData(m_InputValues->NodeTypesDataPath);
```

Comment:
Perhaps I am not understanding, but unless more than the size is being manipulated It should just be as simple as a resize? Otherwise why is an array being created as int_8 then grabbed immediately after as Int32?